src/compiler/program.ts

```diff
@@ -2722,9 +2587,12 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             filesByName.set(path, filesByName.get(oldFile.path));
         });
 
+        const isConfigIdentical = oldOptions.configFile && oldOptions.configFile === options.configFile
+            || !oldOptions.configFile && !options.configFile && !optionsHaveChanges(oldOptions, options, optionDeclarations);

```

Comment:
I wanted to suggest

```suggestion
        // Reuse if both have an identical config file,
        // or neither have a config but the options have changed.
        const isConfigIdentical = oldOptions.configFile === options.configFile && (
            options.configFile || !optionsHaveChanges(oldOptions, options, optionDeclarations))
```

But now the comment is really explaining what the current code is more clearly doing. That said, when do you end up in a situation where neither has a configFile? Inferred projects?

Reply:
Yeah, possibly—I was also wondering about `tsc --watch` without `-p` but I honestly don’t know for sure